gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactorJobRunner.java

```diff
@@ -75,57 +98,73 @@
   protected final FileSystem fs;
   protected final FsPermission perm;
   protected final boolean deduplicate;
+  protected final double priority;
   protected final EventSubmitter eventSubmitter;
 
-  protected MRCompactorJobRunner(State jobProps, FileSystem fs) {
-    this.jobProps = jobProps;
-    this.topic = jobProps.getProp(ConfigurationKeys.COMPACTION_TOPIC);
-    this.inputPath = new Path(jobProps.getProp(ConfigurationKeys.COMPACTION_JOB_INPUT_DIR));
-    this.outputPath = new Path(jobProps.getProp(ConfigurationKeys.COMPACTION_JOB_DEST_DIR));
-    this.tmpPath = new Path(jobProps.getProp(ConfigurationKeys.COMPACTION_JOB_TMP_DIR));
+  private volatile Policy policy = Policy.DO_NOT_PUBLISH_DATA;
+  private volatile Status status = Status.RUNNING;
+
+  protected MRCompactorJobRunner(Dataset dataset, FileSystem fs, Double priority) {
+    this.dataset = dataset;
+    this.jobProps = dataset.jobProps();
```

Comment:
Yes this will change. I think it makes sense that a compaction dataset has a compaction job props, and it's better than adding another Map in `MrCompactor` from dataset to jobprops.